README.md

```diff
@@ -28,15 +28,86 @@ go-jsonschema \
 
 ## Stability definition
 
-Before reaching 1.0, each minor version change is equivalent to major version change. That is, there are no guarantees about compatibility and all changes are permitted. As of 1.0, we provide the following stability guarantees:
+**NOTICE**: The stability definitions are applicable after this repository publishes a stable release (1.0.0). Currently, this repository is experimental and makes no stability guarantees. 
 
-- For major version: No guarantees.
-- For minor versions: TBD
+Stability definition consists of the following sections:
 
-Allowable changes:
+* [Objectives](#objectives): Overview of the motivation behind stability.
+* [Guarantees and allowed changes](#guarantees-and-allowed-changes): Specific details on allowed and disallowed changed.
+* [Applicability](#applicability): Limits of stability definitions, including experimental features and extension points.
+* [File format](#file-format): The `file_format` property and implementation behavior when schema versions are not aligned.
 
-- For major versions: All changes are permitted.
-- For minor versions: TBD
+### Objectives
+
+The primary objective of stability is to protect users from breaking changes. That is, users providing configuration conforming to a particular stable version of the schema should be able to reliably upgrade minor and patch versions without risk that their configuration becomes invalid, or that the interpretation changes.
```

Comment:
This assumes semver versioning. We should specify this, right?

Reply:
Yes, good call. I've added this all the way at the top, so that sections can reference `MAJOR`, `MINOR`, `PATCH` where needed.